tensorflow/mnist.py

```diff
@@ -10,140 +10,136 @@
+
+        fc_weights = weight_variable(DTYPE, [hidden_dim, 10])
+        fc_bias = bias_variable(DTYPE, [10])
+        logits = tf.matmul(reshape, fc_weights) + fc_bias
+        prediction = tf.nn.softmax(logits)
+
+        one_hot_labels = tf.one_hot(labels, depth=10)
+        cost = -tf.reduce_sum(tf.log(prediction) * one_hot_labels, [1])
+        avg_cost = tf.reduce_mean(cost)
+
+        correct = tf.equal(tf.argmax(prediction, 1), labels)
+        accuracy = tf.reduce_mean(tf.cast(correct, tf.float32))
+        g_accuracy = tf.metrics.accuracy(labels, tf.argmax(prediction, axis=1))
+
+        opt = tf.train.AdamOptimizer(
+            learning_rate=0.001, beta1=0.9, beta2=0.999)
+        train_op = opt.minimize(avg_cost)
+        # train_op = tf.train.AdamOptimizer(1e-4).minimize(avg_cost)
+
+    train_reader = paddle.batch(
+        paddle.dataset.mnist.train(), batch_size=args.batch_size)
     test_reader = paddle.batch(
-        paddle.dataset.mnist.test(), batch_size=BATCH_SIZE)
-    for batch_id, data in enumerate(test_reader()):
-        images_data = np.array(
-            map(lambda x: np.transpose(x[0].reshape([1, 28, 28]), axes=[1,2,0]), data)).astype("float32")
-        labels_data = np.array(map(lambda x: x[1], data)).astype("int64")
-        _, loss, acc, g_acc = sess.run(
-            [train_op, avg_cost, accuracy, g_accuracy],
-            feed_dict={images: images_data,
-                       labels: labels_data})
-    return g_acc[1]
-
-
-config = tf.ConfigProto(
-    intra_op_parallelism_threads=1, inter_op_parallelism_threads=1)
-with tf.Session(config=config) as sess:
-    init_g = tf.global_variables_initializer()
-    init_l = tf.local_variables_initializer()
-    sess.run(init_g)
-    sess.run(init_l)
-    for pass_id in range(PASS_NUM):
-        pass_start = time.time()
-        for batch_id, data in enumerate(train_reader()):
+        paddle.dataset.mnist.test(), batch_size=args.batch_size)
+
+    def eval_test():
+        for batch_id, data in enumerate(test_reader()):
             images_data = np.array(
                 map(lambda x: np.transpose(x[0].reshape([1, 28, 28]), axes=[1,2,0]), data)).astype("float32")
             labels_data = np.array(map(lambda x: x[1], data)).astype("int64")
-            start = time.time()
+
             _, loss, acc, g_acc = sess.run(
                 [train_op, avg_cost, accuracy, g_accuracy],
                 feed_dict={images: images_data,
                            labels: labels_data})
-            end = time.time()
-
-            print("pass=%d, batch=%d, loss=%f, error=%f, elapse=%f" %
-                  (pass_id, batch_id, loss, 1 - acc, (end - start) / 1000))
-        pass_end = time.time()
-        test_avg_acc = eval_test()
-        print("pass=%d, training_avg_accuracy=%f, test_avg_acc=%f, elapse=%f" %
-              (pass_id, g_acc[1], test_avg_acc, (pass_end - pass_start) / 1000))
+        return g_acc[1]
```

Comment:
Be careful here,  tf.metrics.accuracy will accumulate the acc. Please make sure to reset the accumulating variables each time doing validate. You can refer https://github.com/tensorflow/tensorflow/issues/4814#issuecomment-328338662

Reply:
Done. Thx!